code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/process/LoadBalancerServiceActivatePostListener.java

```diff
@@ -0,0 +1,120 @@
+import io.cattle.platform.core.model.ServiceExposeMap;
+import io.cattle.platform.engine.handler.HandlerResult;
+import io.cattle.platform.engine.handler.ProcessPostListener;
+import io.cattle.platform.engine.process.ProcessInstance;
+import io.cattle.platform.engine.process.ProcessState;
+import io.cattle.platform.process.common.handler.AbstractObjectProcessLogic;
+import io.cattle.platform.servicediscovery.api.constants.ServiceDiscoveryConstants;
+import io.cattle.platform.servicediscovery.api.dao.ServiceConsumeMapDao;
+import io.cattle.platform.servicediscovery.service.ServiceDiscoveryService;
+import io.cattle.platform.util.type.Priority;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.inject.Inject;
+
+public class LoadBalancerServiceActivatePostListener extends AbstractObjectProcessLogic implements ProcessPostListener,
+        Priority {
+
+    @Inject
+    ServiceDiscoveryService sdService;
+
+    @Inject
+    LoadBalancerTargetDao targetDao;
+
+    @Inject
+    ServiceConsumeMapDao consumeMapDao;
+
+    @Inject
+    LoadBalancerService lbService;
+
+    @Override
+    public String[] getProcessNames() {
+        return new String[] { ServiceDiscoveryConstants.PROCESS_SERVICE_ACTIVATE };
+    }
+
+    @Override
+    public HandlerResult handle(ProcessState state, ProcessInstance process) {
+        Service service = (Service) state.getResource();
+        if (!service.getKind().equalsIgnoreCase(ServiceDiscoveryConstants.KIND.LOADBALANCERSERVICE.name())) {
+            return null;
+        }
+        
+        LoadBalancer lb = objectManager.findOne(LoadBalancer.class, LOAD_BALANCER.SERVICE_ID,
+                service.getId(), LOAD_BALANCER.REMOVED, null);
+        
+        List<Long> consumedInstancesIds = getConsumedInstancesIds(service);
+        List<Long> lbTargetIds = getLoadBalancerTargetInstancesIds(lb);
+
+        addMissingTargets(lb, consumedInstancesIds, lbTargetIds);
+        removeExtraTargets(lb, consumedInstancesIds, lbTargetIds);
+
+        return null;
+    }
+
+    protected void removeExtraTargets(LoadBalancer lb, List<Long> consumedInstancesIds, List<Long> lbTargetIds) {
+        List<Long> targetInstancesToRemove = new ArrayList<>();
+        targetInstancesToRemove.addAll(lbTargetIds);
+        targetInstancesToRemove.removeAll(consumedInstancesIds);
+        for (Long toRemove : targetInstancesToRemove) {
+            lbService.removeTargetFromLoadBalancer(lb, toRemove);
+        }
+    }
+
+    protected void addMissingTargets(LoadBalancer lb, List<Long> consumedInstancesIds, List<Long> lbTargetIds) {
+        List<Long> targetInstanceToAdd = new ArrayList<>();
+        targetInstanceToAdd.addAll(consumedInstancesIds);
+        targetInstanceToAdd.removeAll(lbTargetIds);
+
+        for (Long toAdd : targetInstanceToAdd) {
+            lbService.addTargetToLoadBalancer(lb, toAdd);
+        }
+    }
+
+    protected List<Long> getConsumedInstancesIds(Service service) {
```

Comment:
I have some performance concerns here since I'm thinking we can probably replace it with an SQL query in one of the Dao classes.  But I don't mind tackling this later.